VideoWeb/VideoWeb/Helpers/HearingLayoutService.cs

```diff
@@ -0,0 +1,94 @@
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using VideoApi.Client;
+using VideoApi.Contract.Requests;
+using VideoWeb.Common.Caching;
+using VideoWeb.Common.Models;
+using VideoWeb.EventHub.Hub;
+using VideoWeb.EventHub.Services;
+
+namespace VideoWeb.Helpers
+{
+    public class HearingLayoutService : IHearingLayoutService
+    {
+        private readonly ILogger<HearingLayoutService> _logger;
+        private readonly IVideoApiClient _videoApiClient;
+        private readonly IConferenceCache _conferenceCache;
+        private readonly IHearingLayoutCache _hearingLayoutCache;
+        private readonly IHubContext<EventHub.Hub.EventHub, IEventHubClient> _hubContext;
+
+        public HearingLayoutService(ILogger<HearingLayoutService> logger, IVideoApiClient videoApiClient, IConferenceCache conferenceCache, IHearingLayoutCache hearingLayoutCache, IHubContext<EventHub.Hub.EventHub, IEventHubClient> hubContext)
+        {
+            _logger = logger;
+            _videoApiClient = videoApiClient;
+            _conferenceCache = conferenceCache;
+            _hearingLayoutCache = hearingLayoutCache;
+            _hubContext = hubContext;
+        }
+
+        private async Task<Conference> GetConferenceFromCache(Guid conferenceId)
+        {
+            return await _conferenceCache.GetOrAddConferenceAsync(conferenceId, async () => await _videoApiClient.GetConferenceDetailsByIdAsync(conferenceId));
+        }
+
+        private async Task SetHearingLayoutInCache(Guid conferenceId, HearingLayout newLayout)
+        {
+            await _hearingLayoutCache.WriteToCache(conferenceId, newLayout);
+        }
+
+        private async Task<HearingLayout?> GetHearingLayoutFromCache(Guid conferenceId)
+        {
+            return await _hearingLayoutCache.ReadFromCache(conferenceId);
+        }
+
+        public async Task<HearingLayout?> GetCurrentLayout(Guid conferenceId)
+        {
+            try
+            {
+                var conference = await GetConferenceFromCache(conferenceId);
+                var hearingLayout = await GetHearingLayoutFromCache(conferenceId);
+                return hearingLayout ?? conference.GetRecommendedLayout();
```

Comment:
Some of the conditional blocks in the c# side of things seem like an easy scenario to write a test for. It’s not a total deal breaker but I do think it’s one to cover to make sure the code returns what is expected